arches_for_science/pkg/extensions/functions/multicard_resource_descriptor.py

```diff
@@ -0,0 +1,42 @@
+from arches.app.functions.primary_descriptors import AbstractPrimaryDescriptorsFunction
+from arches.app.models.system_settings import settings
+
+from django.utils.translation import get_language, gettext as _
+
+details = {
+    "functionid": "00b2d15a-fda0-4578-b79a-784e4138664b",
+    "name": "Multi-card Resource Descriptor",
+    "type": "primarydescriptors",
+    "description": "Configure resource descriptors from multiple cards",
+    "defaultconfig": {
+        "descriptor_types": {
+            "name": {"string_template": ""},
+            "map_popup": {"string_template": ""},
+            "description": {"string_template": ""},
+        }
+    },
+    "classname": "MulticardResourceDescriptor",
+    "component": "views/components/functions/multicard-resource-descriptor",
+}
+
+
+class MulticardResourceDescriptor(AbstractPrimaryDescriptorsFunction):
+    """Implemented in the database via triggers on tiles table.
+    
+    This implementation just fetches the calculated result from the db."""
+
+    def get_primary_descriptor_from_nodes(self, resource, config, context=None, descriptor="name"):
+        # TODO: patch core arches to send the appropriate descriptor argument, e.g. "name" or "description" etc.
+        result = ""
+        requested_language = context.get("language", None) if context else None
+
+        lookup_language = requested_language or get_language() or settings.LANGUAGE_CODE
+        try:
+            result = resource.descriptors[lookup_language][descriptor]
```

Comment:
@jacobtylerwalls  - Would this work instead? That way we don't need to add the descriptor type to the call signature.
```
result = resource.descriptors[lookup_language][config["type"]]
```

Reply:
Thanks @bferguso, that's an interesting idea that I'm open to.

My thought at the moment is that adding a `type` key to the config object is duplicative, though, and potentially more work to implement.

I happen to be adding a new descriptor function in this PR, but suppose I weren't -- I'd need to write a data migration to reshape the config objects to provide that missing key, when all of the knowledge is already in the python file.